### PR TITLE
Add TypeScript external access APIs

### PR DESCRIPTION
**Pull request by tmat**
Add some external access APIs: various extension methods, diagnostic related APIs and a few others.

### CODE REVIEW

**Comment by tmat** [comment]
@sharwell PTAL. TypeScript is using a lot of internal APIs and it'd be a lot of work to shim all of them. 
I'm adding some that are related to diagnostic analyzers and a few simple ones.

**Comment by tmat** [comment]
Corresponding internal PR : https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS/pullrequest/224432

**Comment by tmat** [comment]
@jasonmalinowski @sharwell

File: src/EditorFeatures/Core/ExternalAccess/VSTypeScript/Api/VSTypeScriptEditorFormattingServiceWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (ryzngard) [inline src/EditorFeatures/Core/ExternalAccess/VSTypeScript/Api/VSTypeScriptEditorFormattingServiceWrapper.cs]: nit: This is the wrong copyright now, but I don't know if there's some overarching plan to do passes to fix

File: src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptSemanticChangeNotificationServiceWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+using Microsoft.CodeAnalysis.Notification;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
+{
+    internal readonly struct VSTypeScriptSemanticChangeNotificationServiceWrapper
+    {
+        private readonly ISemanticChangeNotificationService _underlyingObject;
+
+        public VSTypeScriptSemanticChangeNotificationServiceWrapper(ISemanticChangeNotificationService underlyingObject)
```
Comment (JoeRobich) [inline src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptSemanticChangeNotificationServiceWrapper.cs]: What value is this if nothing can be read from it?

File: src/EditorFeatures/Core/ExternalAccess/VSTypeScript/Api/VSTypeScriptEditorFormattingServiceWrapper.cs (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/Core/ExternalAccess/VSTypeScript/Api/VSTypeScriptEditorFormattingServiceWrapper.cs]: Good point.

File: src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptSemanticChangeNotificationServiceWrapper.cs (hunk 2 above)
Comment (tmat) [reply src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptSemanticChangeNotificationServiceWrapper.cs]: Good catch. This is not used. Will remove.

**Comment by tmat** [comment]
> TypeScript will export an implementation of IVSTypeScriptDiagnosticAnalyzerImplementation and everything in these changes will work?

Yes.

**Pull request merged**